common/db_store/db.py

```diff
@@ -0,0 +1,120 @@
+from enum import Enum
+import pymongo
+from abc import ABC, abstractmethod
+from common.db_store.error import DBError
+
+
+class DB(ABC):
+
+    @abstractmethod
+    def open_connection(self):
```

Comment:
def open(self)
def close(self)

Pass connection parameters in constructor.

Reply:
Added open & close interfaces.

Reply:
what about connection parameters ?

Reply:
While initializing we're passing endpoint & database information.

Reply:
please add init as discussed.

Reply:
Added '__init__' method in both abstract classes. Also added TODO for singleton design pattern ( Same in LLD as well ).